### PR TITLE
Feature/block state definition

### PR DESCRIPTION
**Pull request by jackz-jones**
1、定义了基本的block、tx、state、receipt数据结构，以及对应的pb。
2、单纯针对共识模块来的，会尽可能简单点，后续根据需要再调整或者增加相应的字段。

### CODE REVIEW

**Comment by vjoke** [comment]
Use bytes for all the big.Int object in proto files.

**Comment by jackz-jones** [comment]
@vjoke 
> Use bytes for all the big.Int objects in proto files.

done

File: core/types/block.proto [hunk 1]
```diff
@@ -16,6 +16,27 @@ message BlockHeaderMsg {
     bytes Nonce = 11;
 }
 
+message TransactionMsg {
+
+    // account nonce
+    uint64 AccountNonce = 1;
```
Comment (git-hulk) [inline core/types/block.proto]: AccountNonce => Nonce

File: core/types/block.proto [hunk 2]
```diff
@@ -16,6 +16,27 @@ message BlockHeaderMsg {
...
+
+    // receiver of this tx
```
Comment (git-hulk) [inline core/types/block.proto]: the receipt of the transaction

File: core/types/block.proto (hunk 1 above)
Comment (jackz-jones) [reply core/types/block.proto]: done

File: core/types/block.proto (hunk 2 above)
Comment (jackz-jones) [reply core/types/block.proto]: It's the receiver of tx, has changed param name:
 Recipient => Receiver

**Pull request merged**